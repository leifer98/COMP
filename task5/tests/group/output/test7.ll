
declare i32 @scanf(i8*, ...)
declare i32 @printf(i8*, ...)
declare void @exit(i32)
@.int_specifier_scan = constant [3 x i8] c"%d\00"
@.int_specifier = constant [4 x i8] c"%d\0A\00"
@.str_specifier = constant [4 x i8] c"%s\0A\00"
define i32 @readi(i32) {
%ret_val = alloca i32
%spec_ptr = getelementptr [3 x i8], [3 x i8]* @.int_specifier_scan, i32 0, i32 0
call i32 (i8*, ...) @scanf(i8* %spec_ptr, i32* %ret_val)
%val = load i32, i32* %ret_val
ret i32 %val
}
define void @printi(i32) {
%spec_ptr = getelementptr [4 x i8], [4 x i8]* @.int_specifier, i32 0, i32 0
call i32 (i8*, ...) @printf(i8* %spec_ptr, i32 %0)
ret void
}
define void @print(i8*) {
%spec_ptr = getelementptr [4 x i8], [4 x i8]* @.str_specifier, i32 0, i32 0
call i32 (i8*, ...) @printf(i8* %spec_ptr, i8* %0)
ret void
}
define i32 @getfive() {
Visiting Return Node
%t0 = zext i8 5 to i32
}
define void @main() {
%t1 = call i32 @getfive()
%x = alloca i32 
store i32 %t1, i32* %x
%t2 = load i32, i32* %x
call void @printi(i32 %t2)
ret void
}
