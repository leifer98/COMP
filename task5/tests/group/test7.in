byte getfive()
{
    return 5b;
}
void main()
{
    int x = getfive();
    printi(x);
}

