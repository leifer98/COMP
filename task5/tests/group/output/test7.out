lli: <stdin>:32:11: error: '%t1' defined with type 'i8' but expected 'i32'
store i32 %t1, i32* %x
          ^
